packages/gitbook/src/lib/api.ts

```diff
@@ -87,21 +87,41 @@ const cacheTtl_1day = {
     ttl: 24 * 60 * 60,
 };
 
-const apiSyncStorage = new AsyncLocalStorage<GitBookAPI>();
+const apiSyncStorage = new AsyncLocalStorage<GitBookAPIWithContextKey>();
 
 export const DEFAULT_API_ENDPOINT = process.env.GITBOOK_API_URL ?? 'https://api.gitbook.com';
 
+export class GitBookAPIWithContextKey extends GitBookAPI {
+
+    #contextKey?: string;
+
+    constructor(options?: ConstructorParameters<typeof GitBookAPI>['0'] & {
+        /** Context key matching the hash of the claims included in the visitor JWT token */
+        contextKey?: string;
+    }) {
+        const { contextKey, ...apiClientOpts} = options || {};
+        super(apiClientOpts);
+
+        this.#contextKey = options?.contextKey;
+    }
+
+    get contextKey() {
+        return this.#contextKey;
+    }
+}
+
 /**
  * Create a new API client with a token.
  */
-export function apiWithToken(apiToken: string): GitBookAPI {
+export function apiWithToken(apiToken: string, contextKey?: string): GitBookAPIWithContextKey {
```

Comment:
```suggestion
export function apiWithToken(apiToken: string, contextKey: string | undefined): GitBookAPIWithContextKey {
```

Optional argument are easy to miss, we should avoid them